artichoke-backend/src/extn/core/string/trampoline.rs

```diff
@@ -601,7 +601,32 @@ pub fn byteindex(
     } else {
         None
     };
-    interp.try_convert(s.index(needle, offset))
+    interp.try_convert(s.byteindex(needle, offset))
+}
+
+pub fn byterindex(
+    interp: &mut Artichoke,
+    mut value: Value,
+    mut substring: Value,
+    offset: Option<Value>,
+) -> Result<Value, Error> {
+    #[cfg(feature = "core-regexp")]
+    if let Ok(_pattern) = unsafe { Regexp::unbox_from_value(&mut substring, interp) } {
+        return Err(NotImplementedError::from("String#byteindex with Regexp pattern").into());
```

Comment:
It looks like this error message should say `String#byterindex`.

Reply:
opps thanks, fixed in 2b7662f0d0d1c4bf695e9f2262af61f17da26f22